bootstrap/config/config_test.go

```diff
@@ -92,3 +108,237 @@ func TestGetSecretPathsChanged(t *testing.T) {
 		})
 	}
 }
+
+func TestLoadCommonConfig(t *testing.T) {
+	serviceKey := "test-service"
+	getAccessToken := func() (string, error) {
+		return "", nil
+	}
+	serviceConfig := ConfigurationMockStruct{
+		Writable: WritableInfo{
+			LogLevel: "INFO",
+		},
+		Registry: config.RegistryInfo{
+			Host: "localhost",
+			Port: 8500,
+			Type: "consul",
+		},
+	}
+
+	appConfig := ConfigurationMockStruct{
+		Writable: WritableInfo{
+			StoreAndForward: StoreAndForwardInfo{
+				Enabled:       false,
+				RetryInterval: "5m",
+				MaxRetryCount: 10,
+			},
+		},
+		Trigger: TriggerInfo{
+			Type: "edgex-messagebus",
+		},
+	}
+
+	deviceConfig := ConfigurationMockStruct{
+		Writable: WritableInfo{
+			Telemetry: config.TelemetryInfo{
+				Metrics: map[string]bool{"EventsSent": false, "ReadingsSent": true},
+			},
+		},
+	}
+
+	tests := []struct {
+		Name                 string
+		serviceConfig        *ConfigurationMockStruct
+		serviceType          string
+		serviceTypeConfig    *ConfigurationMockStruct
+		providerClientErr    error
+		isAlive              bool
+		isCommonConfigReady  []byte
+		commonConfigReadyErr error
+		getConfigErr         error
+		expectedErr          error
+	}{
+		{"Valid - core service", &serviceConfig, config.ServiceTypeOther, nil,
+			nil, true, []byte("true"), nil, nil, nil},
+		{"Valid - app service", &serviceConfig, config.ServiceTypeApp, &appConfig,
+			nil, true, []byte("true"), nil, nil, nil},
+		{"Valid - device service", &serviceConfig, config.ServiceTypeDevice, &deviceConfig,
+			nil, true, []byte("true"), nil, nil, nil},
+		{"Invalid - config provider not alive", &serviceConfig, config.ServiceTypeOther, nil,
+			nil, false, []byte("false"), nil, nil, errors.New("configuration provider is not available")},
```

Comment:
Please create error used in code that can be reference here so you are not making a dup error. Same for other errors below.

Reply:
done